### PR TITLE
Removed plenty of eprintfs in rz_core

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/core/rtr_http.c [hunk 1]
```diff
@@ -132,16 +132,12 @@ static int rz_core_rtr_http_run(RzCore *core, int launch, int browse, const char
...
-	// WHY
-	rz_config_set (core->config, "scr.html", "true");
-#endif
 	rz_config_set_i(core->config, "scr.color", COLOR_MODE_DISABLED);
 	rz_config_set(core->config, "asm.bytes", "false");
 	rz_config_set(core->config, "scr.interactive", "false");
-	eprintf("Starting http server...\n");
-	eprintf("open http://%s:%d/\n", host, atoi(port));
-	eprintf("rizin -C http://%s:%d/cmd/\n", host, atoi(port));
+	RZ_LOG_WARN("core: Starting http server...\n");
```
Comment (ret2libc) [inline librz/core/rtr_http.c]: RZ_LOG_INFO in this case? This does not seem a warning or a problematic thing, just an info.

File: librz/core/rtr.c [hunk 2]
```diff
@@ -1021,7 +1021,7 @@ RZ_API void rz_core_rtr_cmds(RzCore *core, const char *port) {
 		goto err_socket;
 	}
 
-	eprintf("Listening for commands on port %s\n", port);
+	RZ_LOG_WARN("core: listening for commands on port %s\n", port);
```
Comment (ret2libc) [inline librz/core/rtr.c]: INFO here as well?

File: librz/core/rtr.c [hunk 3]
```diff
@@ -705,43 +705,43 @@ RZ_API void rz_core_rtr_add(RzCore *core, const char *_input) {
...
 		if (!rz_socket_connect_udp(fd, host, port, timeout)) { // TODO: Use rap.ssl
 			core->num->value = 1;
-			eprintf("Error: Cannot connect to '%s' (%s)\n", host, port);
+			RZ_LOG_ERROR("core: Cannot connect to '%s' (%s)\n", host, port);
 			rz_socket_free(fd);
 			return;
 		}
 		core->num->value = 0;
-		eprintf("Connected to: %s at port %s\n", host, port);
+		RZ_LOG_DEBUG("core: Connected to: %s at port %s\n", host, port);
```
Comment (ret2libc) [inline librz/core/rtr.c]: info?

Comment (ret2libc) [reply librz/core/rtr.c]: i mean also the others that are similar in this file

File: librz/core/casm.c [hunk 4]
```diff
@@ -797,7 +795,8 @@ static RzList *rz_core_asm_back_disassemble(RzCore *core, ut64 addr, int len, ut
 		rz_asm_set_pc(core->rasm, current_instr_addr);
 		current_instr_len = next_buf_pos - current_buf_pos;
 		current_instr_len = rz_asm_disassemble(core->rasm, &op, buf + current_buf_pos, current_instr_len);
-		IFDBG {
+#if 0
```
Comment (XVilka) [inline librz/core/casm.c]: Maybe we need `RZ_LOG_TRACE()` for such code pieces.

File: librz/core/cil.c [hunk 5]
```diff
@@ -172,7 +172,7 @@ RZ_API void rz_core_analysis_esil_init_mem(RZ_NONNULL RzCore *core, RZ_NULLABLE
 	esil->stack_fd = rz_io_fd_open(core->io, uri, RZ_PERM_RW, 0);
 	if (!(stack_map = rz_io_map_add(core->io, esil->stack_fd, RZ_PERM_RW, 0LL, addr, size))) {
 		rz_io_fd_close(core->io, esil->stack_fd);
-		eprintf("Cannot create map for tha stack, fd %d got closed again\n", esil->stack_fd);
+		RZ_LOG_ERROR("core: cannot create map for tha stack, fd %d got closed again\n", esil->stack_fd);
```
Comment (XVilka) [inline librz/core/cil.c]: `tha` -> `the`

File: librz/core/cio.c [hunk 6]
```diff
@@ -9,13 +9,13 @@ RZ_API int rz_core_setup_debugger(RzCore *r, const char *debugbackend, bool atta
 	RzIODesc *fd = r->file ? rz_io_desc_get(r->io, r->file->fd) : NULL;
 
 	p = fd ? fd->data : NULL;
-	rz_config_set_i(r->config, "cfg.debug", 1);
```
Comment (XVilka) [inline librz/core/cio.c:12]: nice catch.

File: librz/core/cmd/cmd_debug.c [hunk 7]
```diff
@@ -829,13 +830,13 @@ static int dump_maps(RzCore *core, int perm, const char *filename) {
...
 			if (!buf) {
-				eprintf("Cannot allocate 0x%08" PFMT64x " bytes\n", map->size);
+				RZ_LOG_ERROR("core: Cannot allocate 0x%08" PFMT64x " bytes\n", map->size);
 				free(buf);
 				/// XXX: TODO: read by blocks!!1
 				continue;
 			}
 			if (map->size > MAX_MAP_SIZE) {
-				eprintf("Do not dumping 0x%08" PFMT64x " because it's too big\n", map->addr);
+				RZ_LOG_ERROR("core: Do not dumping 0x%08" PFMT64x " because it's too big\n", map->addr);
```
Comment (XVilka) [inline librz/core/cmd/cmd_debug.c]: Remove `do`.

File: librz/core/cmd/cmd_debug.c [hunk 8]
```diff
@@ -2474,14 +2476,14 @@ RZ_IPI int rz_cmd_debug(void *data, const char *input) {
...
-							eprintf("Cannot open those alias files\n");
+							RZ_LOG_ERROR("core: Cannot open those alias files\n");
 						}
 						free(a);
 						free(b);
 					}
 					free(arg);
 				} else {
-					eprintf("Usage: dif $a $b  # diff two alias files\n");
+					RZ_LOG_ERROR("core: Usage: dif $a $b  # diff two alias files\n");
```
Comment (XVilka) [inline librz/core/cmd/cmd_debug.c:2486]: `dif/diff`?

File: librz/core/cmd/cmd_debug.c [hunk 9]
```diff
@@ -2652,10 +2656,11 @@ RZ_IPI int rz_cmd_debug(void *data, const char *input) {
 							bytes, bytes_len,
 							0);
 					} else {
-						eprintf("Invalid hexpairs\n");
+						RZ_LOG_ERROR("core: Invalid hexpairs\n");
 					}
-				} else
-					eprintf("Injection opcodes so long\n");
+				} else {
+					RZ_LOG_ERROR("core: Injection opcodes so long\n");
```
Comment (XVilka) [inline librz/core/cmd/cmd_debug.c:2662]: `too long`?

File: librz/core/cmd/cmd_print.c [hunk 10]
```diff
@@ -4887,7 +4887,7 @@ RZ_IPI int rz_cmd_print(void *data, const char *input) {
 				const bool json = input[2] == 'j'; // ps+j
 				ut64 bitness = rz_config_get_i(core->config, "asm.bits");
 				if (bitness != 32 && bitness != 64) {
-					eprintf("Error: bitness of %" PFMT64u " not supported\n", bitness);
+					RZ_LOG_ERROR("core: bitness of %" PFMT64u " not supported\n", bitness);
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c]: Maybe `%s bits aren't supported`?

**Review by XVilka — CHANGES_REQUESTED** [review]
Significant work.

File: librz/core/cmd/cmd_debug.c (hunk 8 above)
Comment (wargio) [reply librz/core/cmd/cmd_debug.c:2486]: it is the command name

File: librz/core/casm.c (hunk 4 above)
Comment (wargio) [reply librz/core/casm.c]: that piece of code for me can be deleted.

Comment (XVilka) [reply librz/core/casm.c]: Fine for me, just remove it then please

File: librz/core/cmd/cmd_debug.c (hunk 8 above)
Comment (XVilka) [reply librz/core/cmd/cmd_debug.c:2486]: Yes, I figured. Looking up at the code, it seems both `dif` and `diff` are valid commands.

**Comment by XVilka** [comment]
This test new output looks better, actually:
```
[XX] db/cmd/cmd_visual initial Visual view
RZ_NOPLUGINS=1 /usr/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e scr.interactive=true
e scr.columns=80
e scr.rows=12
< q; V
?e
' bins/elf/ls
-- stdout
--- expected
+++ actual
@@ -9,5 +9,4 @@
 0x00005b40  488d 3d21 c701 0048 8d35 1ac7 0100 4829  H.=!...H.5....H)           
 0x00005b50  fe48 c1fe 0348 89f0 48c1 e83f 4801 c648  .H...H..H..?H..H           
 0x00005b60  d1fe 7414 488b 053d c401 0048 85c0 7408  ..t.H..=...H..t.           
-0x00005b70  ffe0 660f 1f44 0000 c30f 1f80 0000 0000  ..f..D..........           25h
-
+0x00005b70  ffe0 660f 1f44 0000 c30f 1f80 0000 0000  ..f..D..........           25h


-- stderr
1000;1006;1015h1000;1006;1015l1049l47l
```
Same with
```
[XX] db/cmd/cmd_ps ps+ libc++ 16bit error
RZ_NOPLUGINS=1 /usr/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e asm.arch=x86
e asm.bits=16
f obj.empty_str 12 @ 0x0804a064
ps+ @ obj.empty_str
' =
-- stderr
--- expected
+++ actual
@@ -1,1 +1,1 @@
-ERROR: core: bitness of 16 not supported
+ERROR: core: 16 bits are not supported by this command
```

**Comment by XVilka** [comment]
```
../librz/core/core.c:3306:12: error: unused variable 'b' [-Werror,-Wunused-variable]
        RzBuffer *b = NULL;
                  ^
1 error generated.
```

**Comment by XVilka** [comment]
```
../librz/core/rtr.c: In function ‘rz_core_rtr_add’:
../librz/core/rtr.c:712:8: error: unused variable ‘n’ [-Werror=unused-variable]
  712 |    int n = rz_socket_rap_client_open(fd, file, 0);
      |        ^
cc1: all warnings being treated as errors
```

**Comment by XVilka** [comment]
Macros related message:
```

[XX] db/cmd/cmd_macros arity errors 3
RZ_NOPLUGINS=1 /home/runner/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '(foo1 fst; ?e foo1 ${fst})
.(foo1)
' =
-- stderr
--- expected
+++ actual
@@ --1,4 +-1,4 @@
 ERROR: Macro 'foo1' expects 1 args, not 0
-core: Error while executing command: .(foo1)
+ERROR: core: Error while executing command: .(foo1)
```

**Comment by wargio** [comment]
Sorry for the delay on fixing this

**Review by XVilka — APPROVED** [review]

**Pull request merged**